test/acceptance/mixins/logger.acceptance.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Application, inject} from '@loopback/core';
+import {RestComponent, RestServer, get, param} from '@loopback/rest';
+import {sinon, Client, createClientForHandler} from '@loopback/testlab';
+import {LoggerMixin, Logger, LogArgs} from '../../..';
+
+describe('logger.mixin (acceptance)', () => {
+  // tslint:disable-next-line:no-any
+  let app: any;
```

Comment:
When `app` is any, IDEs are not able to offer any meaningful autocompletion and the compiler cannot verify correct usage. I am proposing to move the definition of `LoggerApplication` out of `createApp` so that we can use it here. 

A simpler alternative (wich may not work, IDK), is to use `LoggerMixin(Application)` as the type of this variable.

```ts
let app: LoggerMixin(Application);
```

Reply:
This unfortunately doesn't work. I even tried using `typeof` on this as well as `<>`. I think any works well here because multiple mixins can be combined and it becomes difficult to nail down the final shape of the Mixin. 

An interesting thing to note however is that we can use the `LoggerApplication` as the type if the class is moved outside of the `createApplication` function which is what I've done & think this is the approach most users will take as well. 

--- 

I did however come across this while doing some research: https://github.com/Microsoft/TypeScript/pull/12114

I haven't tried any of the generic mappings and from reading I'm not sure it's possible to create a generic enough type for multiple mixins applied to a class.

Reply:
> An interesting thing to note however is that we can use the LoggerApplication as the type if the class is moved outside of the createApplication function which is what I've done & think this is the approach most users will take as well.

👍